http-verbs-common/src/main/scala/uk/gov/hmrc/http/HeaderCarrier.scala

```diff
@@ -72,13 +72,20 @@ case class HeaderCarrier(
   def headersForUrl(config: HeaderCarrier.Config)(url: String): Seq[(String, String)] = {
     val isInternalHost = config.internalHostPatterns.exists(_.pattern.matcher(new URL(url).getHost).matches())
 
-    if (isInternalHost)
-      headers(HeaderNames.explicitlyIncludedHeaders ++ config.headersAllowlist) ++
+    val hdrs =
+      if (isInternalHost)
+        headers(HeaderNames.explicitlyIncludedHeaders ++ config.headersAllowlist) ++
+          config.userAgent.map("User-Agent" -> _).toSeq ++
+          extraHeaders
```

Comment:
I have tried to document in the README. There really is only a few ways to create a HeaderCarrier:
`HeaderCarrierConverter.fromRequest`, `HeaderCarrierConverter.fromRequestAndSession` and `HeaderCarrier()`.
Then, additional headers should be provided via the VERB function.
Given the history of the library, e.g. previously one repo for non-play stuff, and another repo for play dependent parts, the package structure doesn't allow for making anything package private (`uk.gov.hmrc.http` and `uk.gov.hmrc.play.http`). I don't want to update this now, since it's a huge breaking change.

Reply:
The problem is that people are using `hc.copy(otherHeaders = xxx)`. Take a look at this for [example](https://github.com/hmrc/mobile-tax-credits-renewal/blob/e34b00a86d146b8221fadb485dbe5af389f6b38d/app/uk/gov/hmrc/mobiletaxcreditsrenewal/controllers/LiveLegacyMobileTaxCreditsRenewalController.scala#L178). When they do this for otherHeaders, they lose the path that is added by default.

Reply:
We can't stop them without changing all the package structure. The intention of this ticket is to at least document how they should be using the library.